.github/workflows/gradle-build-deploy.yml

```diff
@@ -0,0 +1,58 @@
+name: Gradle Build Deploy
+
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+    branches:
+    - main
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.9
+    - name: Set up Java
+      uses: actions/setup-java@v2
+      with:
+        java-version: 11
+        distribution: adopt
+    - name: Gradle Build
+      run: ./gradlew build test site
+    - name: Upload Build Artifact
+      uses: actions/upload-artifact@v2
+      with:
+        name: website
+        path: build/site
+        
+  deploy:
+    needs:
+    - release
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push' && github.repository == 'jameshughes89/cs102'
+    concurrency: deploying
+    steps: 
+    - uses: actions/checkout@v2
+    - uses: actions/download-artifact@v2
```

Comment:
Remove checkout, the repo code isn't actually used. Also, prefer to name the download step.